splinterd/tests/admin/circuit_lifecycle.rs

```diff
@@ -0,0 +1,665 @@
+        .with_circuit_management_type("test_circuit")
+        .with_application_metadata(b"test_data")
+        .with_comments("test circuit")
+        .with_display_name("test_circuit")
+        .build()
+        .unwrap();
+    create_circuit_message
+        .into_proto()
+        .expect("Unable to get proto from CreateCircuit")
+}
+
+/// Test that a 2-party circuit may be created on a 2-node network.
+///
+/// 1. Create and submit a `CircuitCreateRequest` from the first node
+/// 2. Wait until the proposal is available to the second node, using `list_proposals`
+/// 3. Verify the same proposal is available on each node
+/// 4. Create and submit a `CircuitProposalVote` from the second node to accept the proposal
+/// 5. Wait until the circuit is available on the first node, using `list_circuits`
+/// 6. Verify the same circuit is available to each node
+#[test]
+#[ignore]
+pub fn test_2_party_circuit_lifecycle() {
+    // Start a 2-node network
+    let mut network = Network::new()
+        .with_default_rest_api_variant(RestApiVariant::ActixWeb1)
+        .add_nodes_with_defaults(2)
+        .unwrap();
+    // Create a context in order to produce private keys for the nodes
+    let context = Secp256k1Context::new();
+    // Collect the node information to be used to populate the payloads
+    let mut node_info = HashMap::new();
+    // Get the node and node's client for the first node
+    let node_a = network.node(0).unwrap();
+    let node_a_client = node_a.admin_service_client();
+    let node_a_signer = context.new_signer(context.new_random_private_key());
+    // Using `node_a` here for the first node as a placeholder
+    node_info.insert(
+        "node_a".to_string(),
+        format!("http://localhost:{}", node_a.rest_api_port()),
+    );
+    // Get the node and node's client for the second node
+    let node_b = network.node(1).unwrap();
+    let node_b_client = node_b.admin_service_client();
+    let node_b_signer = context.new_signer(context.new_random_private_key());
+    // Using `node_b` here for the second node as a placeholder
+    node_info.insert(
+        "node_b".to_string(),
+        format!("http://localhost:{}", node_b.rest_api_port()),
+    );
+    let circuit_id = "ABCDE-01234";
+    // Create the `CircuitManagementPayload` to be sent to a node
+    let circuit_payload_bytes =
+        make_create_circuit_payload(&circuit_id, "node_a", node_info.clone(), node_a_signer);
+    // Submit the `CircuitManagementPayload` to the first node
+    let res = node_a_client.submit_admin_payload(circuit_payload_bytes);
+    println!("{:?}", res);
+    assert!(res.is_ok());
+
+    // Wait for the proposal to be committed for the second node
+    let proposal_b;
+    let start = Instant::now();
+    loop {
+        if Instant::now().duration_since(start) > Duration::from_secs(60) {
+            panic!("Failed to detect proposal in time");
+        }
+        let proposals = node_b_client.list_proposals(None, None).unwrap().data;
```

Comment:
All of the unwraps should be expects to give better idea of what happened if there is a failure.

Reply:
Updated most of the unwraps to expects to better explain (kept some unwraps where expect seemed unnecessary and added comments as to why the unwrap is used instead of expect)